cmd/agent/install_mac_os.sh

```diff
@@ -284,7 +284,7 @@ if grep -E 'api_key:( APIKEY)?$' "$etc_dir/datadog.yaml" > /dev/null 2>&1; then
     fi
     printf "\n\033[34m* Restarting the Agent...\n\033[0m\n"
     # systemwide installation is stopped at this point and will be started later on
-    if [ "$systemdaemon_install" != true ]; then
+    if [ "$systemdaemon_install" == true ]; then
```

Comment:
Could you explain this a bit more ? I understand the `=` -> `==` but this one is also reversing the logic

Reply:
MB read to fast, you explained it in the issue:

> I believe the condition is incorrect here: https://github.com/DataDog/datadog-agent/blob/main/cmd/agent/install_mac_os.sh#L288
> Since I didn't specify DD_SYSTEMDAEMON_INSTALL=true, systemdaemon_install will be false. And the logic here tries to stop a launchd service that doesn't exist. There's nothing in the script that sets that variable [aside from reading the value of the daemon install env var](https://github.com/DataDog/datadog-agent/blob/main/cmd/agent/install_mac_os.sh#L39-L42). I think the fix is just flipping the logic.